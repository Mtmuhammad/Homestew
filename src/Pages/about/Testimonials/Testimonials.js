import React from "react";
import "./Testimonials.scss";

const Testimonials = () => {
  return (
    <>
      <hr />
      <div className="testimonials">
        <div className="row d-flex justify-content-center">
          <div className="col-md-10 col-xl-8 text-center">
            <h3 className="mb-4">Testimonials</h3>
            <p className="mb-5 pb-2 mb-md-5 pb-md-0">
              Lorem ipsum dolor sit amet, consectetur adipisicing elit. Fugit,
              error amet numquam iure provident voluptate esse quasi, veritatis
              totam voluptas nostrum quisquam eum porro a pariatur veniam.
            </p>
          </div>
        </div>

        <div
          style={{ marginTop: "3rem", marginBottom: "15rem" }}
          className="row text-center"
        >
          <div className="col-md-4 mt-5 mb-5 mb-md-0">
            <div className="d-flex justify-content-center mb-4">
              <img
                src="https://randomuser.me/api/portraits/men/80.jpg"
                className="rounded-circle shadow-1-strong"
                alt="Tim Perry"
              />
            </div>
            <h5 className="mb-3">Tim Perry</h5>
            <h6 className="mb-3">Web Developer</h6>
            <p className="px-xl-3">
              <i className="fas fa-quote-left pe-2"></i>Lorem ipsum dolor sit
              amet, consectetur adipisicing elit. Quod eos id officiis hic
              tenetur quae quaerat ad velit ab hic tenetur.
            </p>
            <ul className="list-unstyled d-flex justify-content-center mb-0">
              <li>
                <i className="fas fa-star fa-lg text-warning"></i>
              </li>
              <li>
                <i className="fas fa-star fa-lg text-warning"></i>
              </li>
              <li>
                <i className="fas fa-star fa-lg text-warning"></i>
              </li>
              <li>
                <i className="fas fa-star fa-lg text-warning"></i>
              </li>
              <li>
                <i className="fas fa-star-half-alt fa-lg text-warning"></i>
              </li>
            </ul>
          </div>
          <div className="col-md-4 mb-5 mb-md-0">
            <div className="d-flex mt-5 justify-content-center mb-4">
              <img
                src="https://randomuser.me/api/portraits/women/75.jpg"
                className="rounded-circle shadow-1-strong"
                alt="Letitia Powell"
              />
            </div>
            <h5 className="mb-3">Letitia Powell</h5>
            <h6 className="mb-3">Graphic Designer</h6>
            <p className="px-xl-3">
              <i className="fas fa-quote-left pe-2"></i>Ut enim ad minima
              veniam, quis nostrum exercitationem ullam corporis suscipit
              laboriosam, nisi ut aliquid commodi.
            </p>
            <ul className="list-unstyled d-flex justify-content-center mb-0">
              <li>
                <i className="fas fa-star fa-lg text-warning"></i>
              </li>
              <li>
                <i className="fas fa-star fa-lg text-warning"></i>
              </li>
              <li>
                <i className="fas fa-star fa-lg text-warning"></i>
              </li>
              <li>
                <i className="fas fa-star fa-lg text-warning"></i>
              </li>
              <li>
                <i className="fas fa-star fa-lg text-warning"></i>
              </li>
            </ul>
          </div>
          <div className="col-md-4 mb-0">
            <div className="d-flex mt-5 justify-content-center mb-4">
              <img
                src="https://mdbcdn.b-cdn.net/img/Photos/Avatars/img%20(9).webp"
                className="rounded-circle shadow-1-strong"
                alt="John Smith"
              />
            </div>
            <h5 className="mb-3">John Smith</h5>
            <h6 className="mb-3">Marketing Specialist</h6>
            <p className="px-xl-3">
              <i className="fas fa-quote-left pe-2"></i>At vero eos et accusamus
              et iusto odio dignissimos ducimus qui blanditiis praesentium
              voluptatum deleniti atque corrupti.
            </p>
            <ul className="list-unstyled d-flex justify-content-center mb-0">
              <li>
                <i className="fas fa-star fa-lg text-warning"></i>
              </li>
              <li>
                <i className="fas fa-star fa-lg text-warning"></i>
              </li>
              <li>
                <i className="fas fa-star fa-lg text-warning"></i>
              </li>
              <li>
                <i className="fas fa-star fa-lg text-warning"></i>
              </li>
              <li>
                <i className="far fa-star fa-lg text-warning"></i>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </>
  );
};

export default Testimonials;
