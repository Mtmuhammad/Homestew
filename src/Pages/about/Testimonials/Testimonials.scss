@use "../../../Styles/abstracts/mixins" as *;
@use "../../../Styles/abstracts/variables" as *;

.testimonials {
  margin-top: 8rem;

  & img {
    height: 150px;
    width: 150px;

    @include respond(xsphone) {
      height: 120px;
      width: 120px;
    }
    @include respond(tiny) {
      height: 100px;
      width: 100px;
    }
  }

  & h3 {
    font-size: 4.7rem;
    font-weight: 700;
    letter-spacing: 0.03em;
  }

  & h5 {
    font-size: 3.7rem;
    font-weight: 700;
    letter-spacing: 0.03em;
  }

  & h6 {
   font-size: 2rem;
   color: $primary-green;
   letter-spacing: 0.03em;
 }
  & p {
    font-size: 1.8rem;
    margin-top: 1.4rem;
    line-height: 1.6;
  }
  & .fa-quote-left {
   color: $primary-green;
  }
}
