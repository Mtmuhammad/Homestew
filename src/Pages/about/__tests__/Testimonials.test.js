import React from "react";
import { render, screen } from "@testing-library/react";
import Testimonials from "../Testimonials/Testimonials";

describe("smoke and snapshot tests", () => {
  test("should render without crashing", async () => {
    render(<Testimonials />);
  });

  test("should match snapshot", async () => {
    const { asFragment } = render(<Testimonials />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("should render heading", () => {
  test("should render main heading", () => {
    render(<Testimonials />);
    const heading = screen.getByRole("heading", { name: /testimonials/i });
    expect(heading).toBeInTheDocument();
    expect(heading).toHaveClass("mb-4");
  });
});

test("should render images", () => {
  render(<Testimonials />);
  const images = screen.getAllByRole("img");
  expect(images.length).toBe(3);
  expect(images[0]).toHaveClass("rounded-circle", "shadow-1-strong");
});
