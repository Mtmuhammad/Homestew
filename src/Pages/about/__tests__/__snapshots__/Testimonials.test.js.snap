// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <hr />
  <div
    class="testimonials"
  >
    <div
      class="row d-flex justify-content-center"
    >
      <div
        class="col-md-10 col-xl-8 text-center"
      >
        <h3
          class="mb-4"
        >
          Testimonials
        </h3>
        <p
          class="mb-5 pb-2 mb-md-5 pb-md-0"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Fugit, error amet numquam iure provident voluptate esse quasi, veritatis totam voluptas nostrum quisquam eum porro a pariatur veniam.
        </p>
      </div>
    </div>
    <div
      class="row text-center"
      style="margin-top: 3rem; margin-bottom: 15rem;"
    >
      <div
        class="col-md-4 mt-5 mb-5 mb-md-0"
      >
        <div
          class="d-flex justify-content-center mb-4"
        >
          <img
            alt="Tim Perry"
            class="rounded-circle shadow-1-strong"
            src="https://randomuser.me/api/portraits/men/80.jpg"
          />
        </div>
        <h5
          class="mb-3"
        >
          Tim Perry
        </h5>
        <h6
          class="mb-3"
        >
          Web Developer
        </h6>
        <p
          class="px-xl-3"
        >
          <i
            class="fas fa-quote-left pe-2"
          />
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quod eos id officiis hic tenetur quae quaerat ad velit ab hic tenetur.
        </p>
        <ul
          class="list-unstyled d-flex justify-content-center mb-0"
        >
          <li>
            <i
              class="fas fa-star fa-lg text-warning"
            />
          </li>
          <li>
            <i
              class="fas fa-star fa-lg text-warning"
            />
          </li>
          <li>
            <i
              class="fas fa-star fa-lg text-warning"
            />
          </li>
          <li>
            <i
              class="fas fa-star fa-lg text-warning"
            />
          </li>
          <li>
            <i
              class="fas fa-star-half-alt fa-lg text-warning"
            />
          </li>
        </ul>
      </div>
      <div
        class="col-md-4 mb-5 mb-md-0"
      >
        <div
          class="d-flex mt-5 justify-content-center mb-4"
        >
          <img
            alt="Letitia Powell"
            class="rounded-circle shadow-1-strong"
            src="https://randomuser.me/api/portraits/women/75.jpg"
          />
        </div>
        <h5
          class="mb-3"
        >
          Letitia Powell
        </h5>
        <h6
          class="mb-3"
        >
          Graphic Designer
        </h6>
        <p
          class="px-xl-3"
        >
          <i
            class="fas fa-quote-left pe-2"
          />
          Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid commodi.
        </p>
        <ul
          class="list-unstyled d-flex justify-content-center mb-0"
        >
          <li>
            <i
              class="fas fa-star fa-lg text-warning"
            />
          </li>
          <li>
            <i
              class="fas fa-star fa-lg text-warning"
            />
          </li>
          <li>
            <i
              class="fas fa-star fa-lg text-warning"
            />
          </li>
          <li>
            <i
              class="fas fa-star fa-lg text-warning"
            />
          </li>
          <li>
            <i
              class="fas fa-star fa-lg text-warning"
            />
          </li>
        </ul>
      </div>
      <div
        class="col-md-4 mb-0"
      >
        <div
          class="d-flex mt-5 justify-content-center mb-4"
        >
          <img
            alt="John Smith"
            class="rounded-circle shadow-1-strong"
            src="https://mdbcdn.b-cdn.net/img/Photos/Avatars/img%20(9).webp"
          />
        </div>
        <h5
          class="mb-3"
        >
          John Smith
        </h5>
        <h6
          class="mb-3"
        >
          Marketing Specialist
        </h6>
        <p
          class="px-xl-3"
        >
          <i
            class="fas fa-quote-left pe-2"
          />
          At vero eos et accusamus et iusto odio dignissimos ducimus qui blanditiis praesentium voluptatum deleniti atque corrupti.
        </p>
        <ul
          class="list-unstyled d-flex justify-content-center mb-0"
        >
          <li>
            <i
              class="fas fa-star fa-lg text-warning"
            />
          </li>
          <li>
            <i
              class="fas fa-star fa-lg text-warning"
            />
          </li>
          <li>
            <i
              class="fas fa-star fa-lg text-warning"
            />
          </li>
          <li>
            <i
              class="fas fa-star fa-lg text-warning"
            />
          </li>
          <li>
            <i
              class="far fa-star fa-lg text-warning"
            />
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
